File: src/java/org/javaturk/dp/ch03/ocp/factory/solution/HR.java
package org.javaturk.dp.ch03.ocp.factory.solution;

import java.util.ArrayList;
import java.util.List;

/**
 * Class that represents Human Resources in a company.
 * 
 * @author akin
 *
 */
public class HR {

	private List<Employee> employees = new ArrayList<Employee>();

	public HR() {
	}

	public List<Employee> getEmployees() {
		return employees;
	}

	public int getNumberOfEmployees() {
		return employees.size();
	}

	public void addNewEmployee(Employee employee) {
		employees.add(employee);
	}
	
	public void listEmployees(){
		System.out.println("All Employees");
		for(Employee employee : employees)
			System.out.println(employee);
	}
}

File: src/c#/Creational/FactoryMethod/FactoryMethod2/HR.cs
using System;
using System.Collections.Generic;

namespace FactoryMethod2
{
    public class HR
    {
        private List<Employee> employees = new List<Employee>();

		public List<Employee> GetEmployees()
		{
			return employees;
		}

		public int GetNumberOfEmployees()
		{
			return employees.Count;
		}

		public void AddNewEmployee(Employee employee)
		{
			employees.Add(employee);
		}

		public void ListEmployees()
		{
			Console.WriteLine("All Employees");
			foreach (Employee employee in employees)
				Console.WriteLine(employee);
		}
    }
}
